lib/sycamore/sycamore/document.py

```diff
@@ -0,0 +1,388 @@
+from abc import abstractmethod, ABC
+from enum import Enum
+
+
+class Category(Enum):
+    Caption = 1
+    Footnote = 2
+    Formula = 3
+    ListItem = 4
+    PageFooter = 5
+    PageHeader = 6
+    Picture = 7
+    SectionHeader = 8
+    Table = 9
+    Text = 10
+    Title = 11
+    Section = 101
+    Group = 102
+
+
+class Node:
```

Comment:
minor: Again we can defer this, but we should probably call this something like DocNode since we already use Node for the query plan.

Reply:
sure, or differentiate by package name.